src/Symfony/Bundle/WebProfilerBundle/Resources/views/Profiler/base_js.html.twig

```diff
@@ -105,6 +106,64 @@
             });
         }
 
+        #createTableSearchFields() {
+            document.querySelectorAll('div.table-with-search-field').forEach((table, i) => {
+                const searchField = document.createElement('input');
+                searchField.type = 'search';
+                searchField.placeholder = 'search...';
+                searchField.id = `table-search-field-${i}`;
+                searchField.classList.add(`table-search-field-input`);
+                searchField.autocapitalize = 'off';
+                searchField.autocomplete = 'off';
+                searchField.autocorrect = 'off';
+                table.insertBefore(searchField, table.firstChild);
+
+                const labelField = document.createElement('label');
+                labelField.htmlFor = `table-search-field-${i}`;
+                labelField.classList.add(`table-search-field-label`);
+                labelField.innerHTML = 'Search inside the contents of the table';
+                table.insertBefore(labelField, table.firstChild);
+
+                searchField.addEventListener('input', () => {
+                    const query = searchField.value.toLowerCase();
+                    table.querySelectorAll('tbody tr').forEach((row) => {
+                        let rowMatchesQuery = false;
+                        for (const cell of row.querySelectorAll('td, th')) {
+                            const cellValue = cell.textContent || cell.innerText;
+                            if (cellValue.toLowerCase().includes(query)) {
+                                rowMatchesQuery = true;
+                                break;
+                            }
+                        }
+
+                        row.style.display = rowMatchesQuery ? '' : 'none';
+                    });
+
+                    // if there are no results and all rows are hidden, show a message to avoid confusion
+                    const allRows = table.querySelectorAll('tbody tr');
+                    const allRowsAreHidden = Array.from(allRows).every(row => 'none' === row.style.display);
+                    const noResultsElement = table.querySelector('.no-results-message');
+                    if (allRowsAreHidden) {
+                        if (null === noResultsElement) {
+                            const noResultsMessage = document.createElement('tr');
+                            noResultsMessage.classList.add('no-results-message');
+                            const noResultsCell = document.createElement('td');
```

Comment:
should we really add this as a row in the table ? This means that after a search with no results found, this row will become part of the rows that can be found with the search.
Maybe we should insert a node in the outside div rather than the table.

Reply:
You are right! I've just changed this to just add a `<p>` in the parent `<div>` element.